python/paddle/jit/dy2static/program_translator.py

```diff
@@ -1189,10 +1182,50 @@ def _build_once(self, cache_key):
                             var.name, var.shape
                         )
                     )
-        if not _in_amp_guard() and not _in_pure_fp16_guard():
-            concrete_program._to_prim()
 
-        return concrete_program, partial_program_from(concrete_program)
+        class PrimHooker(PartialProgramLayerHook):
+            def __init__(self):
+                self.custom_vjps = set()
+                if core._is_fwd_prim_enabled() and core._is_bwd_prim_enabled():
+                    self.custom_vjps = {
+                        op.type
+                        for op in concrete_program.main_program.block(0).ops
+                        if core.has_comp_grad_op_maker(op.type)
+                    }
+
+            def before_append_backward(
+                self, partial_program_layer, forward_program
+            ):
+                if core._is_fwd_prim_enabled():
+                    to_prim(forward_program.block(0), self.custom_vjps)
+                return forward_program
+
+            def after_append_backward(
+                self, partial_program_layer, whole_program, backward_start_idx
+            ):
+                backward_length = (
+                    len(whole_program.block(0).ops) - backward_start_idx
```

Comment:
Add comments to show we need support other block later

Reply:
Done, check block length and raise exception when the length is over 1.